### PR TITLE
Adding ESBJAVA3447PreserveCharsetInContentTypeTestCase

### PR DESCRIPTION
**Pull request by ChamNDeSilva**
Related Public JIRA of Fix : https://wso2.org/jira/browse/ESBJAVA-3447

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            log.error("Error in executing GET request for CharsetAPI",e);
+        }
+        String[] wireResponse = wireServer.getCapturedMessage().split(System.lineSeparator());
+        for (String line : wireResponse) {
+            if (line.contains("Content-Type")) {
+                if (!line.contains("text/xml; charset=UTF-8")) {
+                    Assert.fail("Content-Type header has dropped Charset");
+                }
+            }
+            Assert.assertTrue(true);
```
Comment (indikasampath2000) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java]: Could you please elaborate meaning of `Assert.assertTrue(true);` ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package org.wso2.am.integration.tests.header.util;
```
Comment (ChamNDeSilva) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java (hunk 1 above)
Comment (ChamNDeSilva) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java]: This is added to pass the test when the content-Type header has not dropped the charset.

Comment (indikasampath2000) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java]: How if we can re-write like below? It looks wired when `Assert.assertTrue(true);` Just my opinion. :smile: 

```
        String charSet = "";
        for (String line : wireResponse) {
            if (line.contains("Content-Type")) {
                charSet = line;
                break;
            }
        }
        Assert.assertTrue(charSet.contains("text/xml; charset=UTF-8"), "Content-Type header has dropped Charset");
```

Comment (ChamNDeSilva) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/header/ESBJAVA3447PreserveCharsetInContentTypeTestCase.java]: ture. :) fixed it and commited.

**Pull request merged**